### PR TITLE
Requirements, Status and Competition CRUD operations together with ex…

### PR DESCRIPTION
**Pull request by mthembbu**
…ceptions completed

### CODE REVIEW

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 1]
```diff
@@ -38,62 +38,61 @@ public IEnumerable<Competition> GetAllCompetitions()
...
+                        Competition dbCompetition = _context.Competitions.FirstOrDefault(c => c.Name == Comp.Name);
+                        if (dbCompetition != null)
+                            return new BadRequestObjectResult(Comp.Name + " already exists");
+                    _notificationMessages.CompetitionNotification(_context, Comp);
+                     _context.Competitions.Add(Comp);
+                    _context.SaveChanges();
+                    dbCompetition = _context.Competitions.Last <Competition>();
+                   for (int i =0; i < 3;i++)
+                    {
+                        Requirement R = new Requirement {
```
Comment (kyle-manganyi) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: variable name R is meaningless

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 2]
```diff
@@ -112,43 +110,66 @@ public async Task<IActionResult> UpdateCompetition(int id, [FromBody]Competition
...
             }
-            else {
-                return new BadRequestObjectResult("Invalid competition model!");
+            else
+            {
+                return new BadRequestObjectResult("Invalid requirements model!");
             }
         }
-        public IEnumerable<Requirement> GetAllRequirementsByCompID(int CompID)
+ //##########################################################################################################
```
Comment (kyle-manganyi) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: GC

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 3]
```diff
@@ -112,43 +110,66 @@ public async Task<IActionResult> UpdateCompetition(int id, [FromBody]Competition
...
+        //GET: api/Competition/Requirements/{id}
+        /**returning an array of requirements for a given competition Id
```
Comment (kyle-manganyi) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: GC

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs (hunk 1 above)
Comment (mthembbu) [reply SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: Ngiyadlala, thanks mate

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs (hunk 3 above)
Comment (JurieOberholzer) [reply SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: Well this is actually a description of what is happening. Not ghost code

**Pull request merged**